### DISCUSSION TITLE
std::equal alternative

### DISCUSSION THREAD

**Discussion by meftunca** [jfalcou/eve#1525] [General]
Hi, can I match Eve with values ​​at a specific position in the string? Do you currently have a structure that corresponds to the std::equal function?

I currently have this build

```c++
        if (std::equal(currentChar, currentChar + 4, "true")) {
            tokens.emplace_back(JSON_TRUE, getPosition(), 4);
            currentChar += 3;
            // continue;
            consumeWS();
        }
```

**Comment by DenisYaroshevskiy** [comment]
For comparing just 4 or 5 chars just use memcmp: https://godbolt.org/z/7c9Tobrz3
Compiler knows what it is and will do the right thing.

On gcc they optimise to memcmp, which is why you can still use equals.
However, on libc++ they don't - so I suggest to use memcpy.

eve has equals, mismatch and so on but since the compiler doesn't know them, it wont' get optimised: https://godbolt.org/z/vMnq4e4Gf
